test/csv/interface/test_read.rb

```diff
@@ -32,6 +32,25 @@ def test_foreach
     assert_equal(@rows, rows)
   end
 
+  def test_foreach_stringio
+    s = StringIO.new
+    s << @data
+
+    rows = []
+    CSV.foreach(s, col_sep: "\t", row_sep: "\r\n") do |row|
+      rows << row
+    end
+    assert_equal(@rows, rows)
+  end
+
+  def test_foreach_stringio_with_bom
+    s = StringIO.new
+    s << "\ufeff" # U+FEFF ZERO WIDTH NO-BREAK SPACE
+    s << @data
```

Comment:
Can we simplify this?

```suggestion
    s = StringIO.new("\ufeff#{@data}") # U+FEFF ZERO WIDTH NO-BREAK SPACE
```